### PR TITLE
Case 19798: 3D projected Material Entities and materialRepeat option

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/19798/Enable-Projected-Material-Entity-mapping-mode

Test plan:
- Make a material entity using Create with URL `https://hifi-content.s3.amazonaws.com/samuel/TestMaterial.json`.  You should see a small sphere with the material on it.
- Apply the material to a Cube.  The cube should be textured.  Change the material mapping properties of the material: material mapping position, material mapping scale, and material mapping rotation.  The changes should take effect.
- Deselect "Material Repeat".  At Material Mapping Scales larger than 1, the texture should stop repeating.  For example, if the scale is x: 10 and y: 10, the texture will be smaller, but instead of repeating, you'll just see it once somewhere on the cube.
- (The next two steps are kinda confusing, but basically it'll look like this: https://hifi-content.s3.amazonaws.com/samuel/SamGondelman%20@%20serverless_%20file____~_serverless_tutorial.json%20-%20Build%20dev%2011_8_2018%203_35_45%20PM.mp4)
- Select "Material Repeat" again.  Change the Material Mapping Mode to "3D projected".  You should now be able to scale the material entity.  Orient the material entity so the blue axis in edit is pointing towards the cube.  As you scale it and move it around, the material applied to the cube should react accordingly, as if the material was being projected onto the surface.
- Deselect "Material Repeat".  Again, the material will stop repeating and you'll only see it once (if the material entity is actually pointed at the surface, otherwise you'll see nothing).
- Run the material tests: https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/material

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14953/HighFidelity-Beta-PR14372-5e167a5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14953/HighFidelity-Beta-PR14372-5e167a5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14953/HighFidelity-Beta-Interface-PR14372-5e167a5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14953/HighFidelity-Beta-PR14372-5e167a5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14953/HighFidelity-Beta-Interface-PR14372-5e167a5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14964/HighFidelity-Beta-PR14372-a94c6e3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14964/HighFidelity-Beta-Interface-PR14372-a94c6e3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14964/HighFidelity-Beta-PR14372-a94c6e3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14964/HighFidelity-Beta-Interface-PR14372-a94c6e3.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15005/HighFidelity-Beta-PR14372-215068e.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15010/HighFidelity-Beta-PR14372-b3abd97.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15005/HighFidelity-Beta-PR14372-215068e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15005/HighFidelity-Beta-Interface-PR14372-215068e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15010/HighFidelity-Beta-PR14372-b3abd97.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15010/HighFidelity-Beta-Interface-PR14372-b3abd97.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15010/HighFidelity-Beta-PR14372-b3abd97.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15010/HighFidelity-Beta-Interface-PR14372-b3abd97.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15481/HighFidelity-Beta-PR14372-860d920.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15481/HighFidelity-Beta-PR14372-860d920.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15481/HighFidelity-Beta-Interface-PR14372-860d920.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15482/HighFidelity-Beta-PR14372-860d920.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15482/HighFidelity-Beta-PR14372-860d920.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15482/HighFidelity-Beta-Interface-PR14372-860d920.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15482/HighFidelity-Beta-PR14372-860d920.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15482/HighFidelity-Beta-Interface-PR14372-860d920.exe

File: libraries/entities/src/MaterialEntityItem.cpp [hunk 1]
```diff
@@ -128,6 +133,7 @@ void MaterialEntityItem::debugDump() const {
     qCDebug(entities) << "           material url:" << _materialURL;
     qCDebug(entities) << "  current material name:" << _currentMaterialName.c_str();
     qCDebug(entities) << "  material mapping mode:" << _materialMappingMode;
+    qCDebug(entities) << "        material repeat:" << _materialRepeat;
     qCDebug(entities) << "               priority:" << _priority;
     qCDebug(entities) << "   parent material name:" << _parentMaterialName;
     qCDebug(entities) << "   material mapping pos:" << _materialMappingPos;
```
Comment (AndrewMeadows) [inline libraries/entities/src/MaterialEntityItem.cpp]: For the spectators: All of those changes above represent the boilerplate required to add a new EntityItemProperty to hifi.

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15690/HighFidelity-Beta-PR14372-860d920.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15690/HighFidelity-Beta-Interface-PR14372-860d920.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15690/HighFidelity-Beta-PR14372-860d920.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15690/HighFidelity-Beta-Interface-PR14372-860d920.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by qaAdamSmith82** [comment]
Pre-Merge Ad-Hoc BVT and Nitpick pass completed

**Pull request merged**